core/lib/Cardano/Address/Script.hs

```diff
@@ -266,35 +300,75 @@ foldScript fn zero = \case
 -- | Validate a 'Script', semantically
 --
 -- @since 3.0.0
-validateScript :: Script -> Either ErrValidateScript ()
-validateScript = \case
-    RequireSignatureOf (KeyHash bytes) -> do
-        when (BS.length bytes /= credentialHashSize) $ Left WrongKeyHash
+validateScript :: Script KeyHash -> Either ErrValidateScript ()
+validateScript script = do
+    let validateKeyHash (KeyHash bytes) =
+            when (BS.length bytes /= credentialHashSize) $ Left WrongKeyHash
+    validateScriptWith validateKeyHash script
+
+validateScriptWith
+    :: Eq elem
+    => (elem -> Either ErrValidateScript ())
+    -> Script elem
+    -> Either ErrValidateScript ()
+validateScriptWith validateRequireSignatureOf = \case
+    RequireSignatureOf element ->
+        validateRequireSignatureOf element
 
     RequireAllOf script -> do
-        when (L.null script) $ Left EmptyList
+        when (L.null (omitTimelocks script)) $ Left EmptyList
```

Comment:
Why the `omitTimeLocks` here :thinking: Surely, we can create a script with only timelocks, no ?

Reply:
indeed, from the ledger point of view it can be acceptable, depends on slots used, where it is used, how it related to tx validity...made distinction between required (as understood by ledger) and recommended validation